src/Symfony/Component/Routing/Matcher/RedirectableUrlMatcher.php

```diff
@@ -32,9 +32,9 @@ public function match($pathinfo)
             }
 
             try {
-                parent::match($pathinfo.'/');
+                $parameters = parent::match($pathinfo.'/');
 
-                return $this->redirect($pathinfo.'/', null);
+                return array_replace($parameters, $this->redirect($pathinfo.'/', isset($parameters['_route']) ? $parameters['_route'] : null));
```

Comment:
So `"symfony/routing": "~2.7.21|~2.8.14|^3.1.7"`? I don't see how composer will know to install the latest 2.7 branch then as 2.7.21 is not released yet.